### ISSUE TITLE
Stack Overflow exception throwing from IDE Startup

### ISSUE THREAD

**Issue by gerardmrk** [mono/monodevelop#5817]
Visual Studio for macOS High Sierra consistently crashes when attempting to view a file after opening a solution. Eventually took a look at the logs from `~/Library/Logs/VisualStudio/7.0/Ide.log`. A couple hundred lines down after a few infos and warnings (plus 2-3 errors, but not exceptions), this happens:

```
...
WARNING [2018-08-31 11:17:23Z]: ANDROID: Package loaded from https://dl.google.com/android/repository/addons_list-3.xml [4876:3], with path 'emulator' has no valid archives (Host OS: Mac)
WARNING [2018-08-31 11:17:23Z]: ANDROID: Element 'http://schemas.android.com/repository/android/sites-common/1:site-list' has a duplicate attribute 'http://www.w3.org/2000/xmlns/:common' in document 'https://dl.google.com/android/repository/addons_list-3.xml'. New value will override the old one
WARNING [2018-08-31 11:17:23Z]: ANDROID: Element 'http://schemas.android.com/repository/android/sites-common/1:site-list' has a duplicate attribute 'http://www.w3.org/2000/xmlns/:sdk' in document 'https://dl.google.com/android/repository/addons_list-3.xml'. New value will override the old one
Stack overflow in unmanaged: IP: 0x10c58656b, fault addr: 0x7ffee529bfe8
Stack overflow: IP: 0x10c58656b, fault addr: 0x7ffee5292fe8
Stacktrace:
  at <unknown> <0xffffffff>
  at string.InternalSubString (int,int) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/string.cs:1265
  <...>
  at string.Substring (int,int) [0x00087] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/string.cs:1257
  at string.Substring (int) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/string.cs:1223
  at MonoDevelop.UnitTesting.VsTest.VsTestUnitTest.GetChildNamespace (string) [0x00029] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestUnitTest.cs:147
  at MonoDevelop.UnitTesting.VsTest.VsTestNamespaceTestGroup.AddTest (MonoDevelop.UnitTesting.VsTest.VsTestUnitTest) [0x00000] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestNamespaceTestGroup.cs:67
...
```

It then continues throwing this for the next 10k+ lines:
```
at MonoDevelop.UnitTesting.VsTest.VsTestNamespaceTestGroup.AddTest (MonoDevelop.UnitTesting.VsTest.VsTestUnitTest) [0x00088] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestNamespaceTestGroup.cs:75
```

and eventually outputs a different stacktrace, the first 4 lines:
```
 at MonoDevelop.UnitTesting.VsTest.VsTestNamespaceTestGroup.AddTests (System.Collections.Generic.IEnumerable`1<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase>) [0x00023] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestNamespaceTestGroup.cs:61
  at MonoDevelop.UnitTesting.VsTest.DiscoveredTests.BuildTestInfo (MonoDevelop.UnitTesting.VsTest.VsTestProjectTestSuite) [0x0002a] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/DiscoveredTests.cs:52
  at MonoDevelop.UnitTesting.VsTest.VsTestProjectTestSuite/<OnCreateTests>d__12.MoveNext () [0x0008e] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestProjectTestSuite.cs:97
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore/MoveNextRunner.InvokeMoveNext (object) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1089
```
and the last 10 lines:
```
at GLib.Timeout/TimeoutProxy.HandlerInternal (intptr) [0x00017] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/gtk-sharp-None/glib/Timeout.cs:38
  at (wrapper native-to-managed) GLib.Timeout/TimeoutProxy.HandlerInternal (intptr) [0x00021] in <270e7412760746b39483cb3d37bf0e03>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) Gtk.Application.gtk_main () [0x00008] in <13dfc6c534074eb08a8f382a81e605d5>:0
  at Gtk.Application.Run () [0x00001] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:145
  at MonoDevelop.Ide.IdeApp.Run () [0x00006] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs:456
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions) [0x007e0] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:332
  at MonoDevelop.Ide.IdeStartup.Main (string[],MonoDevelop.Ide.Extensions.IdeCustomizer) [0x0008f] in /Users/vsts/agent/2.139.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:734
  at Xamarin.Startup.MainClass.Main (string[]) [0x00000] in /Users/vsts/agent/2.139.1/work/1/s/md-addins/Xamarin.Startup/Main.cs:11
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) [0x00057] in <9a00c5ecdc8844fdbd530d1bce707f34>:0
```

I've omitted the rest of the stack trace to avoid bloating up the thread with a huge text wall. If it's helpful, I can post the rest of the stack trace or email the log file to a maintainer.

For context, the directory holds 3 solutions (this boilerplate is from ABP Zero):

- `<APP_NAME>.<APP_NAME>.All`
- `<APP_NAME>.<APP_NAME>.Web`
- `<APP_NAME>.<APP_NAME>.Mobile`

It consistently crashes with `All` and `Web`, but only occasionally so for `Mobile`. The IDE only crashes when attempting to open any project files, but stays running when opening other menus like the SDK manager and preferences.

Purging the numerous `bin` and `obj` directories and performing a rebuild does not resolve the issue. Neither does clearing the cache from the `Library/Caches/VisualSudio/7.0/..`.

This only started happening around 5 days ago, after what looks like a significant update for VisualStudio.

> VS bug [#674974](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/674974)

**Comment by mrward** [comment]
Would you be able to share your solution or the project that uses VS Test to help us reproduce the bug?

It does not seem that this particular part of the code has changed recently but I do see a potential for a stackoverflow on [this line](https://github.com/mono/monodevelop/blob/c49c9323e1b71698f3ae69125614c6961a9c168d/main/src/addins/MonoDevelop.UnitTesting/MonoDevelop.UnitTesting.VsTest/VsTestNamespaceTestGroup.cs#L75).

**Comment by gerardmrk** [comment]
It doesn't look like it's explicitly referenced in any `.csproj` file. Searching for `VsTest` in the whole codebase doesn't show anything either. Is there a specific keyword I can use? I'm not actually familiar with Mono; the code was left over by another developer before me and I'm only now starting to hack into the mobile solution.

**Comment by mrward** [comment]
Is there any test projects in the solution? The VS Test discovery will occur if there is a test project that uses a test adapter NuGet package, such as NUnit3TestAdapter, xunit.runner.visualstudio, or MSTest.TestAdapter.

**Comment by gerardmrk** [comment]
Ah gotcha. This is the csproj file for our tests

`<projectDir>/<app>.<app>.Tests.csproj`
```
<Project Sdk="Microsoft.NET.Sdk">
  <Import Project="..\..\common.props"></Import>
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <AssetTargetFallback>$(AssetTargetFallback);portable-net45+win8+wp8+wpa81;</AssetTargetFallback>
    <AssemblyName>APP.APP.Tests</AssemblyName>
    <PackageId>APP.APP.Tests</PackageId>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
    <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\..\src\APP.APP.Application\APP.APP.Application.csproj" />
    <ProjectReference Include="..\..\src\APP.APP.EntityFrameworkCore\APP.APP.EntityFrameworkCore.csproj" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Data.Sqlite" Version="2.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.0.3" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.7.0" />
    <PackageReference Include="xunit" Version="2.3.1" />
    <PackageReference Include="xunit.extensibility.execution" Version="2.3.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
    <PackageReference Include="Shouldly" Version="3.0.0" />
    <PackageReference Include="NSubstitute" Version="3.1.0" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Abp.TestBase" Version="3.8.2" />
  </ItemGroup>
  <ItemGroup>
    <Service Include="{xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx}" />
  </ItemGroup>
  <ItemGroup>
    <None Update="appsettings.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </None>
    <None Update="sqlite3.dll">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
  </ItemGroup>
</Project>
```

**Comment by mrward** [comment]
Would it be possible to get a copy of just that test project and its files?

**Comment by gerardmrk** [comment]
The tests are huge, over 5000k+ LOC collectively, and also riddled with what I think are somewhat sensitive data. For now, I've disabled the `MonoDevelop Assembly Browser` and `MonoDevelop Hex Editor` IDE extensions, and everything's back to normal after rebuilding.

I'd liked to have continued with resolving this issue with you, but given the time constraints and my insufficient experience with Mono (I'm relatively new to the Xamarin ecosystem), I don't think I can pursue this any further. I'm sorry!

From previous googling, it seems like I'm the only one experiencing this stack overflow, so it might even be a problem with local machine setup rather than the tool itself. So feel free to close this issue.

**Comment by mrward** [comment]
Not a problem. I will try to see if I can reproduce this myself. Re-opening this.

**Comment by wilsondy** [comment]
I got this today.  It is real.  Came from nowhere for me.  Did not add tests.

**Comment by mrward** [comment]
@wilsondy Can you attach the IDE log? This particular problem is specific to tests so you may be seeing a different problem.

**Comment by wilsondy** [comment]
I believe it is the same.  These are shortened but if you need the bottom let me know.  They are from two different versions of VS.

[76bug7.6.7.txt](https://github.com/mono/monodevelop/files/2420687/76bug7.6.7.txt)
[bug76_2190.txt](https://github.com/mono/monodevelop/files/2420688/bug76_2190.txt)

**Comment by mrward** [comment]
@wilsondy - Thanks. Yes that looks the same to me. Would you be able to share the project that causes this?

**Comment by rodrmoya** [comment]
Just for the record, can easily replicate this with the xUnit VB.NET template in https://github.com/mono/monodevelop/pull/6222 just building the project shows this stack overflow

**Comment by SayusiAndo** [comment]
I have the same issue, if you need I can send a copy from my code. My solution contains only ~160 xunit tests.

**Comment by mrward** [comment]
This should be fixed in 7.6. Another bug was filed for the same problem and fixed in this pull request for 7.6 - https://github.com/mono/monodevelop/pull/6290
